sklearn/utils/tests/test_testing.py

```diff
@@ -84,10 +90,181 @@ def _raise_ValueError(message):
+
+        # predict alone different
+        t1._same_predict = False
+        assert_array_not_equal(t1.predict(X3), t2.predict(X3))
+        assert_raises(AssertionError, assert_same_model, X3, t1, t2)
+        assert_not_same_model(X3, t1, t2)
+
+        # decision_function alone different
+        t1._same_predict = True
+        t1._same_decision_function = False
+        assert_array_not_equal(t1.decision_function(X3),
+                               t2.decision_function(X3))
+        assert_raises(AssertionError, assert_same_model, X3, t1, t2)
+        assert_not_same_model(X3, t1, t2)
+
+        # predict_proba alone different
+        t1._same_decision_function = True
+        t1._same_predict_proba = False
+        assert_array_not_equal(t1.predict_proba(X3), t2.predict_proba(X3))
+        assert_raises(AssertionError, assert_same_model, X3, t1, t2)
+        assert_not_same_model(X3, t1, t2)
+
+        # transform alone different
+        t1._same_predict_proba = True
+        t1._same_transform = False
+        assert_array_not_equal(t1.transform(X3), t2.transform(X3))
+        assert_raises(AssertionError, assert_same_model, X3, t1, t2)
+        assert_not_same_model(X3, t1, t2)
+
+        # Attributes alone differ
+        t1._same_transform = True
+        t1._same_attributes = False
+        t1.fit(X1, y1)  # Attributes are set inside fit only
+        assert_not_equal(t1.attribute1_, t2.attribute1_)
+        assert_array_not_equal(t1.attribute2_, t2.attribute2_)
+        assert_raises(AssertionError, assert_same_model, X3, t1, t2)
+        assert_not_same_model(X3, t1, t2)
+
+        # All set to True
+        t1._same_attributes = True
+        t1.fit(X1, y1)
+        assert_equal(t1.attribute1_, t2.attribute1_)
+        assert_array_equal(t1.attribute2_, t2.attribute2_)
+        assert_array_equal(t1.predict(X3), t2.predict(X3))
+        assert_array_equal(t1.transform(X3), t2.transform(X3))
+        assert_array_equal(t1.predict_proba(X3), t2.predict_proba(X3))
+        assert_array_equal(t1.decision_function(X3),
+                           t2.decision_function(X3))
+
+        # Any two set to differ
+        t1._same_predict_proba = False
+        t1._same_predict = False
+        assert_raises(AssertionError, assert_same_model, X3, t1, t2)
+        assert_not_same_model(X3, t1, t2)
+
+        # Any one set to False along with differing attributes
+        t1._same_predict_proba = True
+        t1._same_transform = False
+        t1._same_attributes = False
+        t1.fit(X1, y1)  # Attributes are set inside fit only
+        assert_raises(AssertionError, assert_same_model, X3, t1, t1)
+        assert_not_same_model(X3, t1, t2)
+
+        # All set to False
+        t1._same_decision_function = False
+        t1._same_predict_proba = False
+        t1._same_predict = False
+        assert_raises(AssertionError, assert_same_model, X3, t1, t2)
+        assert_not_same_model(X3, t1, t2)
```

Comment:
Actually I would just drop the checks in `test_assert_xsame_model_` tests the `MockClassifier` class and drop the `MockClassifier` class it-self. It just too convoluted and I don't like the reliance on the `np.random` singleton to implement the "not same" case.

I would just keep the checks that use `LinearSVC`.